rasa/core/actions/action.py

```diff
@@ -248,10 +249,17 @@ async def run(
 
         logger.debug(f"Picking response from selector of type {query_key}")
         selected = response_selector_properties[query_key]
-        message = selected[OPEN_UTTERANCE_PREDICTION_KEY]
-        message["template_name"] = selected[FULL_RETRIEVAL_INTENT]
+        possible_messages = selected[OPEN_UTTERANCE_PREDICTION_KEY]["name"]
 
-        return [create_bot_utterance(message)]
+        # Pick a random message from list of candidate messages
+        picked_message_idx = random.randint(0, len(possible_messages) - 1)
+        picked_message = copy.deepcopy(possible_messages[picked_message_idx])
```

Comment:
I know before you were looking into using the NLG. I think here we don't support channel-specific utterances? That is something I think users will expect support for

Reply:
Hmm that's tricky. I would definitely like to make use of the NLG for it which means we need to have the response templates shared between the domain and NLU training data. I'll prefer to keep it out of scope of this PR till we have handled that. I'll mention it in the changelog though.

Reply:
Sounds good, creating an issue for it would be good too :)